common/transaction/src/lib.rs

```diff
@@ -48,15 +48,17 @@ impl<M: ManagedTypeApi> Default for CallData<M> {
 pub struct EthTransaction<M: ManagedTypeApi> {
     pub from: EthAddress<M>,
     pub to: ManagedAddress<M>,
+    pub refund_address: ManagedAddress<M>,
     pub token_id: TokenIdentifier<M>,
     pub amount: BigUint<M>,
     pub tx_nonce: TxNonce,
     pub call_data: ManagedOption<M, ManagedBuffer<M>>,
 }
 
-pub type EthTxAsMultiValue<M> = MultiValue6<
+pub type EthTxAsMultiValue<M> = MultiValue7<
     EthAddress<M>,
     ManagedAddress<M>,
+    ManagedAddress<M>,
```

Comment:
is this the refund address? Why can we not refund in the original, provided address?

Reply:
The original address may be a SC and a withdraw function may not be existent.
Also the transaction may fail because the address is inexistent, frozen account, TransferRole, etc.